### ISSUE TITLE
Support for kodi external player

### ISSUE THREAD

**Issue by arlekyn2** [elgatito/plugin.video.elementum#1053]
Anyone managed to make Elementum work with kodi external player (Just player for example) when using "memory"? For me it just plays the initially downloaded part of the torrent - the buffer, after that it stops (the player stops or rather keeps buffering, no crash of Elementum). It's like the Elementum addon stops downloading right after the external player starts running.

## Steps to Reproduce (for bugs)

1.
2.
3.
4.

## Your Environment

* Version used:

* Environment name and version:

* Operating System and version:

* Link to Kodi debug log:

**Comment by elgatito** [comment]
@arlekyn2 Please, provide a Kodi log file, to see what happens when you start a playback with external player (from the start of Kodi to the playback failure). https://elementum.surge.sh/contribute/log/

**Comment by arlekyn2** [comment]
[kodi.log](https://github.com/user-attachments/files/16323719/kodi.log)

**Comment by elgatito** [comment]
@arlekyn2 I think log only shows start of playback.

I think the problem is that Elementum is checking if Kodi is playing something or stopped playback. There are multiple controls around that.

And in your case there is no playback. 

I wanted to see a moment in the log when Elementum stops downloading and removes the torrent (if that is what you actually see).

**Comment by arlekyn2** [comment]
While I can't say exactly what happens at the program level (was hopping the debug log would bring some light), I can describe what actually happens: 
After i select the torrent, everything goes normally - i get the popup with the available seeds and the initial download begins. When it reaches 100% downloading in that popup, the popup is replaced with the player (in our case, the external player Just player). The player will actually display the video file for the corresponding downloaded piece... let's say 50mb of data. After that, the player just freeze (the image), without showing a "buffering" sign or anything. Same happens in VLC player, as well. And in MX player if i remember well. Also, no Elementum crash notification. 
As additional info, what i noticed: if i press "back" button, I go back into kodi as if no playing was initiated. Starting the stream again , even when selecting "from the beginning", it will actually download the following piece of the torrent, while the first piece is still saved in the memory. So this time it will play more (let's say 100mb instead of 50). Going back into kodi and restarting the torrent will just keep adding whatever data is downloaded in that initial popup.
Unfortunately, google chromecast sucks for not being accessible via wireless usb debugging (the only way I can push the playcorefactory.xml file)... It's a mess even retrieving the kodi.log file (also with adb) so it is not... "lucrative" to keep disconnecting it from the TV and connecting it to the laptop and the other way round for testing/debugging.

**Comment by elgatito** [comment]
@antonsoroko should external player work with Elementum on Kodi 21 and newer?

**Comment by antonsoroko** [comment]
@elgatito no, there is a regression (in my opinion) in v21 and now "play using"  works only on real files. 
https://kodi.moy.su/forum/4-23-2#583
(It has explanation and links to GitHub , I do not want to paste them here, because then this issue will be linked to Kodi repo issue/pr.)

**Comment by elgatito** [comment]
I mean maybe there is no sense to fix this issue with external player if it would not run on newer Kodis.

**Comment by antonsoroko** [comment]
I wrote a question about this in Kodi's Issue 2 weeks ago (and i also asked in PR today), but i have not received an answer yet.
I think we can wait and if they say that they will not fix it or they will not answer - then yes, it does not make sense to add a "obsolete" feature.

**Comment by antonsoroko** [comment]
btw, they returned "external player" for addons in 21.2.
https://kodi.tv/article/kodi-21-2-omega-release/
>Fixed missing context menu entries for plugin-provided items
